### ISSUE TITLE
Broken API key management in DigitalOcean backend

### ISSUE THREAD

**Issue by joepie91** [NixOS/nixops#628]
While trying to get a DigitalOcean test deployment working, I've had a read through the [backend implementation](https://github.com/NixOS/nixops/blob/985886cff4ac09c9cb635e31b45f0cc0dbd84e0a/nixops/backends/digital_ocean.py), and I suspect that there are some issues with it.

The first issue is that the `digitalOcean.authToken` property is never actually stored on the `DigitalOceanState` instance in any way, rather only on the `DigitalOceanDefinition` (as `auth_token`). At the same time, `<DigitalOceanState>.get_auth_token()` attempts to read it from `self.auth_token`, which because of the aforementioned issue doesn't exist.

This means that the authentication token - if not set through an environment variable, for which I created a separate PR (#626) - will always be `None`, resulting in the following error:

```
machine.> creating droplet ...error: No token provided. Please use a valid token
```

The immediate solution would be to do something like `self.auth_token = defn.auth_token` in the `create()` call, but this would result in another issue concerning API credential management; I've filed an issue about that in #627.

cc @teh

**Comment by teh** [comment]
I think my immediate preferred solution for this (not the more abstract #627) would be to only allow the env var. That way there at least is no ambiguity in the code. @joepie91 what do you think?

I don't have a strong opinion on a more general key management, it's not an area where I'm good at.

**Comment by joepie91** [comment]
Yeah, I think that would be a reasonable short-term solution. It'd require a change of the documentation as well, though - currently, the environment variable is only documented in the option (which would obviously disappear).

I'd definitely like to see a well-integrated solution for key management that doesn't rely on environment variables at some point, so that credentials can be easily managed and used alongside the rest of the NixOps configuration, especially when dealing with *multiple* sets of credentials for different networks... but that's probably going to have to involve a change to the core of NixOps itself.
